Interop/Tensorflow/src/main/java/org/tribuo/interop/tensorflow/TensorFlowCheckpointModel.java

```diff
@@ -0,0 +1,188 @@
+import java.util.logging.Logger;
+
+/**
+ * TensorFlow support is experimental, and may change without a major version bump.
+ * <p>
+ * This model encapsulates a simple model with an input feed dict,
+ * and produces a single output tensor.
+ * <p>
+ * If the checkpoint is not available on construction or after deserialisation then the
+ * model is uninitialised. Models can be initialised by calling {@link #initialize} or
+ * {@link #setCheckpointDirectory}.
+ * <p>
+ * This model's serialized form stores the weights in the specified model checkpoint directory.
+ * If you wish to convert it into a model which stores the weights inside the model then
+ * call {@link #convertToNativeModel}.
+ * <p>
+ * It accepts an {@link FeatureConverter} that converts an example's features into a {@link TensorMap}, and an
+ * {@link OutputConverter} that converts a {@link Tensor} into a {@link Prediction}.
+ * <p>
+ * The model's serialVersionUID is set to the major Tensorflow version number times 100.
+ * <p>
+ * N.B. Tensorflow support is experimental and may change without a major version bump.
+ */
+public class TensorFlowCheckpointModel<T extends Output<T>> extends TensorFlowModel<T> implements Closeable {
+    private static final Logger logger = Logger.getLogger(TensorFlowCheckpointModel.class.getName());
+
+    private static final long serialVersionUID = 200L;
+
+    private String checkpointDirectory;
+
+    private String checkpointName;
+
+    private boolean initialized;
+
+    TensorFlowCheckpointModel(String name, ModelProvenance description, ImmutableFeatureMap featureIDMap, ImmutableOutputInfo<T> outputIDMap, GraphDef graphDef, String checkpointDirectory, String checkpointName, int batchSize, String initName, String outputName, FeatureConverter<T> featureConverter, OutputConverter<T> outputConverter) {
+        super(name, description, featureIDMap, outputIDMap, graphDef, batchSize, initName, outputName, featureConverter, outputConverter);
+        this.checkpointDirectory = checkpointDirectory;
+        this.checkpointName = checkpointName;
+        try {
+            session.restore(resolvePath());
+            initialized = true;
+        } catch (TensorFlowException e) {
+            logger.log(Level.WARNING, "Failed to initialise model in directory " + checkpointDirectory, e);
+        }
+    }
+
+    /**
+     * Resolves the path into the format that TensorFlow expects.
+     * @return The TensorFlow checkpoint path.
+     */
+    private final String resolvePath() {
+        return Paths.get(checkpointDirectory,checkpointName).toString();
+    }
+
+    /**
+     * Is this model initialized?
+     * @return True if the model is ready to make predictions.
+     */
+    public boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * Initializes the model.
+     * <p>
+     * Throws {@code TensorFlowException} if it failed to read the checkpoint.
+     */
+    public final void initialize() {
```

Comment:
Do we really want to close the old session if someone calls this twice? If so, maybe we should mention it in the javadocs?

Reply:
Well I guess I could check to see if the path has changed, but over time I suspect there will be more things which will cause a reinitialisation. I've not exposed the session config options, nor the ability to move things between CPU and GPU, but both of those will require a reinit. As the set expands I think it's easier to just have it always reinit. I can document that it does that.

Reply:
I added a note to the javadoc.